### PR TITLE
feat(ui): supported resolution center in data quality

### LINKED ISSUE #13746: Improve DQ Resolution Flow
### Context and Abstract
In 1.1.0 we introduce a resolution flow allowing user to understand the status of a data quality failure. In of 1.1.0 and 1.2.0 users are able to set 3 status:
- `New`: denote a newly failed data quality test
- `Ack`: denote an acknowledge failure (i.e. we are working on solving it)
- `Resolved`: denote a solved data quality failure

in 1.3.0 we will introduce an enhanced flow adding 2 additional status:
- `assign`: allows to assign the resolution to someone else
     - assignee
- `review`: we'll be introducing a new policy `tests resolution approval`, users without this policy will not be able to mark issue as resolved. They will need to set a reviewer. It should include the following fields
    -  reviewer
    -  resolution summary

assigning an assignee or a reviewer should open a task for the user of concern. Closing a task for an assignee result in a review or a resolution (depending on the permission). Closing a reviewer task should result in a resolution. When a task is marked as resolved, a user should be able to choose if they want to mark all the previously failed tests as resolved as well.

We will also prevent users from changing the status of a resolved test case failure. 

### User Stories
- As a user I want to be able to track what DQ issues do I need to solve 
- As a user I want to be able to control who can mark a DQ issue as resolved

### Resolution flow approval diagram
image

Below are the lists of all the permutation for the task status

```
combination ('New', 'Ack')
-------
New -> Ack
Ack -> New

combination ('New', 'Assigned')
-------
New -> Assigned
Assigned -> New

combination ('New', 'InReview')
-------
New -> InReview
InReview -> New

combination ('New', 'Resolved')
-------
New -> Resolved

combination ('Ack', 'Assigned')
-------
Ack -> Assigned
Assigned -> Ack

combination ('Ack', 'InReview')
-------
Ack -> InReview
InReview -> Ack

combination ('Ack', 'Resolved')
-------
Ack  -> Resolved

combination ('Assigned', 'InReview')
-------
Assigned -> InReview
InReview -> Assigned

combination ('Assigned', 'Resolved')
-------
Assigned -> Resolved

combination ('InReview', 'Resolved')
-------
Resolved -> InReview

combination ('New', 'Ack', 'Assigned')
-------
New -> Ack -> Assigned
New -> Assigned -> Ack
Ack -> New -> Assigned
Ack -> Assigned -> New
Assigned -> New -> Ack
Assigned -> Ack -> New

combination ('New', 'Ack', 'InReview')
-------
New -> Ack -> InReview
New -> InReview -> Ack
Ack -> New -> InReview
Ack -> InReview -> New
InReview -> New -> Ack
InReview -> Ack -> New

combination ('New', 'Ack', 'Resolved')
-------
New -> Ack -> Resolved
Ack -> New -> Resolved

combination ('New', 'Assigned', 'InReview')
-------
New -> Assigned -> InReview
New -> InReview -> Assigned
Assigned -> New -> InReview
Assigned -> InReview -> New
InReview -> New -> Assigned
InReview -> Assigned -> New

combination ('New', 'Assigned', 'Resolved')
-------
New -> Assigned -> Resolved
Assigned -> New -> Resolved

combination ('New', 'InReview', 'Resolved')
-------
New -> InReview -> Resolved
InReview -> New -> Resolved

combination ('Ack', 'Assigned', 'InReview')
-------
Ack -> Assigned -> InReview
Ack -> InReview -> Assigned
Assigned -> Ack -> InReview
Assigned -> InReview -> Ack
InReview -> Ack -> Assigned
InReview -> Assigned -> Ack

combination ('Ack', 'Assigned', 'Resolved')
-------
Ack -> Assigned -> Resolved
Assigned -> Ack -> Resolved

combination ('Ack', 'InReview', 'Resolved')
-------
Ack -> InReview -> Resolved
InReview -> Ack -> Resolved

combination ('Assigned', 'InReview', 'Resolved')
-------
Assigned -> InReview -> Resolved
InReview -> Assigned -> Resolved

combination ('New', 'Ack', 'Assigned', 'InReview')
-------
New -> Ack -> Assigned -> InReview
New -> Ack -> InReview -> Assigned
New -> Assigned -> Ack -> InReview
New -> Assigned -> InReview -> Ack
New -> InReview -> Ack -> Assigned
New -> InReview -> Assigned -> Ack
Ack -> New -> Assigned -> InReview
Ack -> New -> InReview -> Assigned
Ack -> Assigned -> New -> InReview
Ack -> Assigned -> InReview -> New
Ack -> InReview -> New -> Assigned
Ack -> InReview -> Assigned -> New
Assigned -> New -> Ack -> InReview
Assigned -> New -> InReview -> Ack
Assigned -> Ack -> New -> InReview
Assigned -> Ack -> InReview -> New
Assigned -> InReview -> New -> Ack
Assigned -> InReview -> Ack -> New
InReview -> New -> Ack -> Assigned
InReview -> New -> Assigned -> Ack
InReview -> Ack -> New -> Assigned
InReview -> Ack -> Assigned -> New
InReview -> Assigned -> New -> Ack
InReview -> Assigned -> Ack -> New

combination ('New', 'Ack', 'Assigned', 'Resolved')
-------
New -> Ack -> Assigned -> Resolved
New -> Assigned -> Ack -> Resolved
Ack -> New -> Assigned -> Resolved
Ack -> Assigned -> New -> Resolved
Assigned -> New -> Ack -> Resolved
Assigned -> Ack -> New -> Resolved

combination ('New', 'Ack', 'InReview', 'Resolved')
-------
New -> Ack -> InReview -> Resolved
New -> InReview -> Ack -> Resolved
Ack -> New -> InReview -> Resolved
Ack -> InReview -> New -> Resolved
InReview -> New -> Ack -> Resolved
InReview -> Ack -> New -> Resolved


combination ('New', 'Assigned', 'InReview', 'Resolved')
-------
New -> Assigned -> InReview -> Resolved
New -> InReview -> Assigned -> Resolved
Assigned -> New -> InReview -> Resolved
Assigned -> InReview -> New -> Resolved
InReview -> New -> Assigned -> Resolved
InReview -> Assigned -> New -> Resolved

combination ('Ack', 'Assigned', 'InReview', 'Resolved')
-------
Ack -> Assigned -> InReview -> Resolved
Ack -> InReview -> Assigned -> Resolved
Assigned -> Ack -> InReview -> Resolved
Assigned -> InReview -> Ack -> Resolved
InReview -> Ack -> Assigned -> Resolved
InReview -> Assigned -> Ack -> Resolved


combination ('New', 'Ack', 'Assigned', 'InReview', 'Resolved')
-------
New -> Ack -> Assigned -> InReview -> Resolved
New -> Ack -> InReview -> Assigned -> Resolved
New -> Assigned -> Ack -> InReview -> Resolved
New -> Assigned -> InReview -> Ack -> Resolved
New -> InReview -> Ack -> Assigned -> Resolved
New -> InReview -> Assigned -> Ack -> Resolved
New -> Resolved -> Ack -> Assigned -> InReview
New -> Resolved -> Ack -> InReview -> Assigned
New -> Resolved -> Assigned -> Ack -> InReview
New -> Resolved -> Assigned -> InReview -> Ack
New -> Resolved -> InReview -> Ack -> Assigned
New -> Resolved -> InReview -> Assigned -> Ack
Ack -> New -> Assigned -> InReview -> Resolved
Ack -> New -> InReview -> Assigned -> Resolved
Ack -> Assigned -> New -> InReview -> Resolved
Ack -> Assigned -> InReview -> New -> Resolved
Ack -> InReview -> New -> Assigned -> Resolved
Ack -> InReview -> Assigned -> New -> Resolved
Ack -> Resolved -> New -> Assigned -> InReview
Ack -> Resolved -> New -> InReview -> Assigned
Ack -> Resolved -> Assigned -> New -> InReview
Ack -> Resolved -> Assigned -> InReview -> New
Ack -> Resolved -> InReview -> New -> Assigned
Ack -> Resolved -> InReview -> Assigned -> New
Assigned -> New -> Ack -> InReview -> Resolved
Assigned -> New -> InReview -> Ack -> Resolved
Assigned -> Ack -> New -> InReview -> Resolved
Assigned -> Ack -> InReview -> New -> Resolved
Assigned -> InReview -> New -> Ack -> Resolved
Assigned -> InReview -> Ack -> New -> Resolved
InReview -> New -> Ack -> Assigned -> Resolved
InReview -> New -> Assigned -> Ack -> Resolved
InReview -> Ack -> New -> Assigned -> Resolved
InReview -> Ack -> Assigned -> New -> Resolved
InReview -> Assigned -> New -> Ack -> Resolved
InReview -> Assigned -> Ack -> New -> Resolved
```

#### Backend
- [x] #14049
- [x] Add `assign` status.
- [ ] ~~Create "resolution approval" permission~~
- [x] #14248
- [x] Make test case resolution status with status "resolved" immutable
- [x] #14331
- [ ] Implement severity failure inference

#### Ingestion
- [ ] ~~Update logic to post in the new flow~~

#### UI
- [ ] Implement Resolution center as per mock

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes #13746

I worked on Incident manager feature
- Added new incident manager feature
- unit test
- cypress test

image
image
image
image

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/ProfilerDashboard/component/data-quality-tab.less [hunk 1]
```diff
@@ -42,5 +42,12 @@
         color: @green-3;
       }
     }
+    &.assigned {
+      background-color: #ffbe0e26;
+      border: 1px solid #d99600;
+      .ant-typography {
+        color: #d99600;
+      }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/ProfilerDashboard/component/data-quality-tab.less:50]: Can we use variable instead hardcoded colors?

File: openmetadata-ui/src/main/resources/ui/src/components/common/SummaryCard/SummaryCard.interface.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@
  *  limitations under the License.
  */
 export interface SummaryCardProps {
-  type?: 'success' | 'failed' | 'aborted';
+  type?: 'success' | 'failed' | 'aborted' | 'assigned' | 'acknowledged' | 'new';
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/common/SummaryCard/SummaryCard.interface.ts:14]: Can we use an ENUM here?

File: openmetadata-ui/src/main/resources/ui/src/pages/TasksPage/shared/Assignees.tsx [hunk 3]
```diff
@@ -36,13 +42,21 @@ const Assignees: FC<Props> = ({
   onChange,
   options,
   disabled,
+  isSingleSelect = false,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/TasksPage/shared/Assignees.tsx:45]: What's `isSingleSelect`?

File: openmetadata-ui/src/main/resources/ui/src/rest/testAPI.ts [hunk 4]
```diff
@@ -99,7 +100,7 @@ export const getTestCaseByFqn = async (
   params?: { fields?: string[] }
 ) => {
   const response = await APIClient.get<TestCase>(
-    `/dataQuality/testCases/name/${fqn}`,
+    `/dataQuality/testCases/name/${getEncodedFqn(fqn)}`,
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/rest/testAPI.ts:103]: Please make sure that we are testing affected areas here

File: openmetadata-ui/src/main/resources/ui/src/components/Task/TaskTab/TaskTab.component.tsx [hunk 5]
```diff
@@ -586,73 +676,178 @@ export const TaskTab = ({
...
+                onTextChange={(value) =>
+                  form.setFieldValue('testCaseFailureComment', value)
+                }
               />
             </Form.Item>
-          ) : (
+          </Form>
+        </Modal>
+      ) : (
+        <Modal
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Task/TaskTab/TaskTab.component.tsx:750]: Let's check if can reduce this changes, Seems concerning to me

File: openmetadata-ui/src/main/resources/ui/src/pages/TasksPage/shared/Assignees.tsx (hunk 3 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/pages/TasksPage/shared/Assignees.tsx:45]: there can be a multi-select option for assignees or we can have single select.

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**